src/h3/_cy/memory.pyx

```diff
@@ -0,0 +1,115 @@
+from libc cimport stdlib
+from cython.view cimport array
+from .h3lib cimport H3int
+
+
+cdef size_t move_nonzeros(H3int* a, size_t n):
+    """ Move nonzero elements to front of array `a` of length `n`.
+    Return the number of nonzero elements.
+
+    Loop invariant: Everything *before* `i` or *after* `j` is "done".
+    Move `i` and `j` inwards until they equal, and exit.
+    You can move `i` forward until there's a zero in front of it.
+    You can move `j` backward until there's a nonzero to the left of it.
+    Anything to the right of `j` is "junk" that can be reallocated.
+
+    | a | b | 0 | c | d | ... |
+            ^           ^
+            i           j
+
+
+    | a | b | d | c | d | ... |
+            ^       ^
+            i       j
+    """
+    cdef:
+        size_t i = 0
+        size_t j = n
+
+    while i < j:
+        if a[j-1] == 0:
+            j -= 1
+            continue
+
+        if a[i] != 0:
+            i += 1
+            continue
+
+        # if we're here, we know:
+        # a[i] == 0
+        # a[j-1] != 0
+        # i < j
+        # so we can swap! (actually, move a[j-1] -> a[i])
+        a[i] = a[j-1]
+        j -= 1
+
+    return i
+
+
+cdef H3int[:] empty_memory_view():
```

Comment:
I know this is just copying an existing function, but can this have a docstring of what this function is used for? A memoryview isn't growable iirc, so not clear to me what the point of an empty memory view is.

Reply:
Ah, ok. I can try to add one.

The point of an empty memory view is that we could have situations where functions return an empty set.

IIRC, this was the cleanest way to do that. The syntax for creating one was pretty ugly otherwise, as far as I could figure it out, so I banished it to this function.

It'd be amazing if someone figured out a cleaner way to do this!